qiskit_ibm_runtime/session.py

```diff
@@ -141,7 +141,8 @@ def run(
         """
 
         options = options or {}
-        options["backend"] = self._backend
+        if "backend" not in options:
+            options["backend"] = self._backend
```

Comment:
Can you raise an exception if a session is already active, has a backend, and the backend passed in `options` doesn't match.